spec/controllers/api/enterprises_controller_spec.rb

```diff
@@ -21,15 +21,33 @@ module Api
       end
 
       describe "creating an enterprise" do
-        let(:australia) { Spree::Country.find_by_name('Australia') }
-        let(:new_enterprise_params) { {enterprise: {name: 'name', email: 'email@example.com', address_attributes: {address1: '123 Abc Street', city: 'Northcote', zipcode: '3070', state_id: australia.states.first, country_id: australia.id } } } }
+        let(:address) { build(:address) }
+
+        before :each do
+          @new_enterprise_params = {
+            enterprise: {
+              name: Faker::Name.name,
+              email: 'email@example.com',
+              address_attributes: {
+                address1: '123 Abc Street',
+                city: 'Northcote',
+                zipcode: '3070',
+                state_id: address.state_id,
+                country_id: address.country_id
+              }
+            },
+            format: :json
+          }
+        end
 
-        it "creates as sells=any when it is not a producer" do
-          spree_post :create, new_enterprise_params
-          response.should be_success
+        it "responds successfully" do
+          spree_post :create, @new_enterprise_params
+          response.status.should eq 201
+        end
 
-          enterprise = Enterprise.last
-          enterprise.sells.should == 'any'
+        it "creates a sells=any when it is not a producer" do
+          spree_post :create, @new_enterprise_params
+          JSON.parse(response.body)["sells"].should eq 'any'
```

Comment:
I saw a test helper somewhere to access the already parsed JSON somewhere... I'll try to find it.